### ISSUE TITLE
Question: How do we handle themes?

### ISSUE THREAD

**Issue by brownjohnf** [resin-io/resin-blog#26]
The current way things are deployed works like this:

1. The builders build the image, which contains the `content` directory from this repo
2. The container is run by the supervisor, and sets the `GHOST_DATA_PATH=/data/content`
3. Ghost connects to a mariadb database running in RDS
4. `/data/content` inside the container maps to `/resin-data/<app-id>/content` on the host
5. `/resin-data` is actually an NFS mount, shared between all devices in the application
  * This allows us to share uploaded content across blog instances

Now, my question is how we want to handle the `themes` folder. Any changes made to the theme there will NOT be propagated to the deployed instances. You'd need to bundle it as a zip and upload it through the admin interface.

There are various ways we can solve this issue, but before I go down any roads, I wanted to run this by you @craig-mulligan

**Comment by craig-mulligan** [comment]
You mean `/content/data` right?

> Any changes made to the theme there will NOT be propagated to the deployed instances. You'd need to bundle it as a zip and upload it through the admin interface.

So if I understand correctly only the db is currently sync'd right? I think thats ideal. I'd like to avoid uploading themes from the admin interface and only using `git push resin` to deploy. It'll ensure we know exactly what src is running on the instances.

**Comment by brownjohnf** [comment]
No, it's almost the opposite of what you said :-).

I mean `/data/content`. Because the blog serves assets that are uploaded and not built into the container, its data (both db and files) have to be shared between instances.

`GHOST_DATA_PATH=/data/content`, `/data` is a bind-mount of `/resin-data/<app-id>` (this is different than normal devices; I've modified the supervisor config), `/resin-data` on the host is an NFS mount shared between all devices in the fleet.

The problem is that changes to themes locally will NOT update the theme running on the blog. If that's the behavior we want, we'll need to either modify the container startup process to copy its theme folder over the deployed one (which is problematic, because then all instances will try to do that), modify ghost to use a theme path separate from its data path (this is my preferred option, but I don't know about complexity), or set up ghost to store all content in s3.

**Comment by craig-mulligan** [comment]
Oh i see haha.. `/data/content` and `content/data` is confusing. Yea I definitely don't want themes to be copied across instances. Let's just put images and data in `/data` using http://support.ghost.org/config/#paths. Shouldn't be to tricky I can PR it in a once I'm finnished with the other one.

**Comment by craig-mulligan** [comment]
Spoke to soon, ghost doesn't allow you to configure seperate paths for data/themes/images it only allows you to set the parent content path. http://support.ghost.org/config/#paths. so won't be so simple

**Comment by craig-mulligan** [comment]
How much control do you have with mounting the volumes? Could you bind mount for each subdirectory? image/data and then ignore themes?

**Comment by brownjohnf** [comment]
Nope. It's already a bit of an antipattern to support the /data bind-mount being different than the resin-supervisor data directory.

**Comment by brownjohnf** [comment]
How hard would it be to fork ghost to add this functionality? We should also look around at what other people using ghost in docker do. I have to say, in general, I'm appalled at the poor design of ghost when it comes to scaling and configuration. Even running behind a load balancer is a pain in the ass because it validates URLs

**Comment by craig-mulligan** [comment]
@brownjohnf I forked ghost and added a themes path. It appears to work locally but when I pushed to staging I hit that knex pool err you hit a while ago. Could you take look?

**Comment by brownjohnf** [comment]
Sorry, I missed this today. I'll take a look on Monday.

**Comment by craig-mulligan** [comment]
No worries 👍 I hope it's not that bug again. :/ I'll try push to a vanilla resin device if a get a sec.

**Comment by brownjohnf** [comment]
@craig-mulligan can you provide more context for this? Where is your fork, where is your integration, etc.?

**Comment by craig-mulligan** [comment]
@brownjohnf the integration is here: https://github.com/resin-io/resin-blog/tree/ghost-fork
and the fork is here https://github.com/craig-mulligan/Ghost/tree/stable.

**Comment by craig-mulligan** [comment]
> This allows us to share uploaded content across blog instances

@brownjohnf looks like it's not a good idea - https://github.com/TryGhost/Ghost/issues/4760

**Comment by brownjohnf** [comment]
Hm, that's concerning. I don't really understand that design, unless they're assuming that we don't have the NFS+mysql setup. What state would be in memory, I wonder?

What do you want to do? We can easily just run one ghost instance (I still want the theme split out + nfs for persistance).
